### PR TITLE
[Feat] Multi-output composited screenshots

### PR DESCRIPTION
**Pull request by Shinyzenith**
I attempted to clean up the code a little bit and introduce some style changes to the same PR. The original author has been credited in the commit sign off.

@mstoeckl @Decodetalkers

Supercedes: https://github.com/waycrate/wayshot/pull/40

### CODE REVIEW

**Review by Decodetalkers — DISMISSED** [review]

File: wayshot/src/wayshot.rs [hunk 1]
```diff
@@ -96,67 +66,121 @@ fn main() -> Result<(), Box<dyn Error>> {
...
+                output,
+                None,
+            )?],
+            None,
+        ),
     };
-
     let extension = if args.is_present("extension") {
         let ext: &str = &args.value_of("extension").unwrap().trim().to_lowercase();
+        log::debug!("Using custom extension: {:#?}", ext);
```
Comment (Decodetalkers) [inline wayshot/src/wayshot.rs:182]: if case is stdout, won't the log be mixed with buffer in stdout which is the part of image?

Comment (Shinyzenith) [reply wayshot/src/wayshot.rs:182]: They'd need to turn on debug flag using `-d`. If you're mixing that with `--stdout` it is bound to disturb the image data.

Comment (Shinyzenith) [reply wayshot/src/wayshot.rs:182]: Actually this is a really good point. We should probably make it such that `--stdout` and `--debug` are in conflict with each other. What do you think? @Decodetalkers

Comment (Decodetalkers) [reply wayshot/src/wayshot.rs:182]: I think it is necessary, from my opinion..

**Comment by Shinyzenith** [comment]
I think once we can figure out the transform issue, this PR will be ready to merge?

**Comment by Shinyzenith** [comment]
Thank you for the comprehensive report. ❤️

**Comment by Shinyzenith** [comment]
I finally got some hardware to test this patch out. My results seem to not work.

Wayshot: 
1688281051-wayshot

grim:
20230702_12h27m11s_grim

**Comment by Decodetalkers** [comment]
> I finally got some hardware to test this patch out. My results seem to not work.
> 
> Wayshot: 1688281051-wayshot
> 
> grim: 20230702_12h27m11s_grim

Is the screen right rotated 90 degree? on my machine it works

**Comment by Shinyzenith** [comment]
It's not rotated at all. @Decodetalkers Are you testing this branch or your branch? I tested with your branch again and it seems to work. With this branch it doesn't. 

I'll try to find the issue. Also would you guys mind if I add you both as repo collaborators, would be helpful to have the patch branches here itself. Let me know if you're interested.

**Comment by Decodetalkers** [comment]
> It's not rotated at all. @Decodetalkers Are you testing this branch or your branch? I tested with your branch again and it seems to work. With this branch it doesn't.
> 
> I'll try to find the issue. Also would you guys mind if I add you both as repo collaborators, would be helpful to have the patch branches here itself. Let me know if you're interested.

I test mime, it works, so I don't have clue why this branch not work.. emm..

I will feel glad if I can help

**Comment by Shinyzenith** [comment]
Nope I'm just stupid. It works fine.

**Comment by Shinyzenith** [comment]
I have written a solution to multi-output transformed screenshots with reference to the commit that @Decodetalkers  made: https://github.com/waycrate/wayshot/pull/41/commits/a9749deedaece0245f7dd76e5e5a41c815216267.

I need to find an ergonomic way to do the same for single output screenshots and I think this PR will be ready.

**Comment by Shinyzenith** [comment]
I have fixed single output transforms handling too. I will do some cleanup and commit it.

**Comment by Shinyzenith** [comment]
@mstoeckl @Decodetalkers Thank you so much for your help, it would be impossible to get this PR finished without you guys. I have completed transform handling and committed it. 

If you guys can leave a review that would help a lot. I'll rebase and merge after the review.

**Review by Decodetalkers — APPROVED** [review]
I think it is good.

**Comment by Shinyzenith** [comment]
Merged to main branch.

**Pull request merged**